src/lib/parquet/src/Flow/Parquet/ParquetFile/Schema/NestedColumn.php

```diff
@@ -17,11 +17,57 @@ final class NestedColumn implements Column
     public function __construct(
         private readonly FlatColumn $root,
         array $columns,
-        private readonly int $maxDefinitionsLevel,
-        private readonly int $maxRepetitionsLevel,
         private ?Column $parent = null
     ) {
         $this->children = $columns;
+
+        foreach ($columns as $column) {
+            if ($column->parent() === null) {
```

Comment:
Is it possible that the column has a different parent?

Reply:
No, but it was possible that the parent was already set due to a misalignment between rebuilding schema from thrift and code directly.